### PR TITLE
feat(metrics): Add size limits for metric envelope items

### PR DESCRIPTION
**Pull request by Dav1dde**
Introduce the missing size limits for the metric envelope items.

### CODE REVIEW

File: relay-config/src/config.rs [hunk 1]
```diff
@@ -604,6 +610,9 @@ impl Default for Limits {
             max_api_chunk_upload_size: ByteSize::mebibytes(100),
             max_profile_size: ByteSize::mebibytes(50),
             max_span_size: ByteSize::mebibytes(1),
+            max_statsd_size: ByteSize::kibibytes(100),
```
Comment (Dav1dde) [inline relay-config/src/config.rs]: Not sure if this is a good limit?

File: relay-config/src/config.rs [hunk 2]
```diff
@@ -604,6 +610,9 @@ impl Default for Limits {
...
+            max_metric_buckets_size: ByteSize::mebibytes(1),
```
Comment (Dav1dde) [inline relay-config/src/config.rs:614]: This was discussed with Jan, apparantely the SDKs already have that limit

File: relay-config/src/config.rs [hunk 3]
```diff
@@ -563,6 +563,12 @@ struct Limits {
     max_profile_size: ByteSize,
     /// The maximum payload size for a span.
     max_span_size: ByteSize,
+    /// The maximum payload size for a stasd metric.
```
Comment (TBS1996) [inline relay-config/src/config.rs]:
```suggestion
    /// The maximum payload size for a statsd metric.
```

**Review by TBS1996 — COMMENTED** [review]
lgtm, we should probably figure out what a good statsd limit is before going forward though

File: relay-config/src/config.rs (hunk 1 above)
Comment (jan-auer) [reply relay-config/src/config.rs]: Let's set 1MB for all of these, DDM sends items larger than 100kB.

**Review by jan-auer — APPROVED** [review]

**Review by olksdr — APPROVED** [review]

**Pull request merged**